### PR TITLE
Added unit tests and set up Travis

### PR DESCRIPTION
**Pull request by calcmogul**
This required refactoring the Task class interface to support pipelining file contents. Bugs were fixed as the unit tests uncovered them. Installation and usage documentation was also added to the README.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+dist: trusty
+language: python
+python:
+  - "3.5"
+
+before_install:
+  - sudo apt-get update -q || true
```
Comment (333fred) [inline .travis.yml]: We don't need sudo, do we? https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise#L451

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+dist: trusty
```
Comment (333fred) [inline .travis.yml]: Why trusty?

File: README.md [hunk 3]
```diff
@@ -9,6 +9,48 @@ This repository contains our style guides for C++ and Java code, various IDE sup
...
 Anything submitted to a wpilibsuite project needs to follow the code style guides outlined here. For details about the style, please see the contributors document [here](CONTRIBUTING.md#coding-guidelines).
 
+## Setup
+
+### Dependencies
+
+- [Python 3.5](https://www.python.org/downloads/)
+- clang-format 3.8 or newer (included with [LLVM](http://llvm.org/releases/download.html))
+
+After cloning this repository, set the `WPI_FORMAT` environment variable to its location. If you would like to use these tools with a new project, copy [formatw.py](formatw.py), [.styleguide](#.styleguide), and [.styleguide-license](#.styleguide-license) into its directory.
```
Comment (333fred) [inline README.md]: What is "its"?

File: .travis.yml (hunk 2 above)
Comment (calcmogul) [inline .travis.yml]: We need at least that to get clang-format 3.8.

File: .travis.yml (hunk 1 above)
Comment (calcmogul) [inline .travis.yml]: I don't know. That's what allwpilib uses.

File: README.md (hunk 3 above)
Comment (calcmogul) [inline README.md]: s/its/the project/. I'll fix that.

File: .travis.yml (hunk 1 above)
Comment (ThadHouse) [inline .travis.yml]: It looks like you currently need sudo support in order to be able to run trusty. And it looks like above we need at least trusty.

**Comment by calcmogul** [comment]
My Windows 8 VM is uncovering bugs that AppVeyor should be finding, but doesn't.

File: .travis.yml (hunk 1 above)
Comment (333fred) [inline .travis.yml]: http://genbattle.bitbucket.org/blog/2016/01/17/c++-travis-ci/

Comment (333fred) [inline .travis.yml]: Or better: http://stackoverflow.com/a/35113462

Comment (calcmogul) [inline .travis.yml]: I'm confused. This repo only has Python. It doesn't need a C++ setup.

Comment (333fred) [inline .travis.yml]: Yes, but part of getting clang 3.8 is clang format.

Comment (calcmogul) [inline .travis.yml]: Are you sure those are packaged together on Ubuntu? That would make the clang-format package redundant.

Comment (333fred) [inline .travis.yml]: See my first link. Sure, the instructions are about getting C++. But strip out the C++ stuff you don't need and use the clang format package from my first link and that should do the trick without relying on Travis's beta container.

Comment (calcmogul) [inline .travis.yml]: That config isn't automatically installing dependencies, not that that's a big deal. However, `clang-format-3.8 : Depends: libstdc++6 (>= 4.9) but 4.6.3-1ubuntu5 is to be installed` https://travis-ci.org/wpilibsuite/styleguide/jobs/160723947

Comment (333fred) [inline .travis.yml]: Add these repos: https://github.com/333fred/travis-clang-test/blob/master/.travis.yml#L5

Comment (calcmogul) [inline .travis.yml]: That worked.

**Comment by calcmogul** [comment]
I figured out the problem. I installed Git for Windows with "clone CRLF, commit LF" set, but it's cloning with LF anyway. Since Python sets os.linesep to "\r\n", newline.py cuts off more characters from the end when trimming newlines than it should.

AppVeyor is doing the correct thing.

**Pull request merged**